Compatibilité Mantis 2.x + améliorations

Adysone opened the pull request:
# Changelog

* Le plugin est maintenant compatible avec Mantis 2.x
* Il est désormais possible de supprimer une règle définie précédemment
* Les note de rappel peuvent être privée (se configure au niveau de la règle)
* Le message de rappel est pris en compte (dans la langue saisie uniquement)
* Les macros `{from_status}`, `{to_status}`, `{reminder_days}`, `{status_days}` sont disponibles pour le message de rappel.
* BUGFIX: L'historique d'un bug n'était pas modifié lors de l'ajout automatique d'une note de rappel

Commit:
[Fix] "<<<<< HEAD" était resté en haut du fichier !

Commit:
UI compatible avec Mantis 2.x, ajout de la fonction de suppression et du nom du projet dans la liste

Commit:
Normalise les requêtes SQL

Commit:
Possibilité de rendre les notes privées

Commit:
Améliore l'UI

Commit:
Le cron est maintenant compatible avec les nouvelles fonctionnalités

Commit:
Traduction en Anglais des nouveaux termes

Commit:
Bump des versions sur le README.md

Commit:
Merge branch 'compatibility-mantisbt-2.x'

nenes25 commented:
Bonjour,

Merci pour votre contribution qui à l'air parfaite et remets ce plugin au goût du jour :)
Je teste rapidement et je merge dans la foulée.

Cordialement,
Hervé

Adysone commented:
Bonjour, merci pour votre réponse rapide 😉 

Edit : il va falloir faire un tag pour la v1.3.x de Mantis comme j'ai mis dans le README 😄

Adysone commented:
Ne mergez pas de suite, y a un bug 🤣 

Mais je ne sais pas comment le corriger, dans l'email automatique de Mantis, il met un lien vers 

http://yourwebsite.com/mantisdir/plugins/AutoChangeStatus/pages/view.php?id=XXX#cYYY

au lieu de 

http://yourwebsite.com/mantisdir/view.php?id=XXX#cYYY

Je vais voir, c'est peut-être l'appel du cron qui ne va pas, mais à part appeler le fichier directement, je ne sais pas comment contourner l'authentification obligatoire. Une idée ?

nenes25 commented:
Je vous avouerais que dans l'immédiat je ne me souviens même plus comment ce plugin marche vu la dernière fois que je l'ai édité ^^

Il y'a des chances que ce problème soit lié à cette ligne :
```
#En cron on push le nom du module
plugin_push_current('AutoChangeStatus');
```
mais elle est nécessaire pour le workflow, je vais essayer de regarder également ;)

Commit:
Modification de la manière d'appeler le cron

Il faut appeler l'URL /plugin.php?page=AutoChangeStatus/cron et non /plugins/AutoChangeStatus/pages/cron.php

Commit:
Update README.md

Commit:
Merge branch 'compatibility-mantisbt-2.x'

Adysone commented:
Pour le bug, c'est bon (cf. Issue https://github.com/Adysone/mantisbt_autochangestatus/issues/3)

nenes25 commented:
OK c'est mergé , merci beaucoup ;)

Adysone commented:
Merci !

nenes25 commented:
Au passage si tu veux mettre te rajouter dans les crédits dans l'entête des fichiers n'hésites pas.
Je viens de voir qu'il faut les mettre à jour ils datent encore de 2016 .
J'ai identifié également une erreur dans mon module de base lorsqu'il n'y aucune entrée de renseignée, je vais voir pour fixer ça

Adysone commented:
Il va y avoir une nouvelle PR dans les prochains jours, et pas sûr que ça soit rétrocompatible avec l'existant... Ça sera à tester.

nenes25 commented:
OK je vais l'attendre alors pour corriger les warning sur l'édition d'un élément :)
J'ai juste fait un update des années de copyright et du changelog.

Adysone commented:
OK, je récupérerai les modifs également.
Pour mon besoin, je ne souhaiterais que les messages de rappel, et pas de mise à jour de statut, je compte donc modifier le comportement du cron, pour que si "actif" est à TRUE, ça change le statut, sinon non, mais ça regarde quand même si "reminder" est à TRUE. Pour désactiver une règle, faut mettre les deux à FALSE. Est-ce que c'est dérangeant pour toi ?

nenes25 commented:
Non pas de soucis c'est une évolution pertinente :)
